lib/matplotlib/ticker.py

```diff
@@ -1821,6 +1822,12 @@ def __init__(self, *args, **kwargs):
             Relax `nbins` and `integer` constraints if necessary to
             obtain this minimum number of ticks.
 
+        *tick_space_heuristic*
```

Comment:
I'd suggest that this just be another option for nbins.  i.e. if nbins is callable it is used to determine the heuristic...

Reply:
That's a great idea!